### ISSUE TITLE
Any news of the current state of the project?

### ISSUE THREAD

**Issue by hanaral** [Suzamax/Aircraft#4]
Elementary OS 6 is right around the corner, and there has been a huge increase in the apps on the appcenter. As there are new, more professional grade apps like Akira on the way, I think that Aircraft would have similar success to Planner.
If funding is required I would not mind chipping in via GitHub Sponsors.

**Comment by ccnmxns** [comment]
I have news!

First of all, sorry for being so absent from this project. Rather busy because University, some side-projects, personal life, and that feeling of being unable to make something good, but I hope that feeling will get the fuck off me soon :smile:
Second: It would be GTK-agnostic. As I said in another issue, I want it to be more "general" and less elementary-based. Most people use GTK but less people use Pantheon than people using GNOME. I do currently use GNOME as my daily-driven desktop.
Third: I'm moving it from Vala to C++ (GTKMM), thus making it easier to use TdLib as it have native C++ libraries, and I feel quite comfortable using C++ rather than Vala. I'll upload ASAP the basis of the new Aircraft.
Also: if I feel this is going somewhere good I'll think about funding. But only if this get at least a MVP, not before. I don't like the idea of being a ripoff.

PS: I'm so bad at designing, if this works I'll try to make an Adwaita-like icon, but I suck with vectorization, coloring and the likes.

**Comment by hanaral** [comment]
Nice to hear! I'd be lying if I said I wasn't a little disappointed that it isn't sticking to the appcenter, but then again it's obv your project. I hope to see more progress soon.
I could give making an icon, but it wouldn't be very Adwaita, since I think they aren't very portable and accessible (i.e try looking at the evince or EoG icon at 48px on a white backdrop).

**Comment by ccnmxns** [comment]
I didn't say it wouldn't be available or not through AppCenter, it 
obviously depends on elementary stuff. I don't think it would be 
approved as it wouldn't be "native" on a elementary standard basis, but 
I want it more portable.

About the icon: I looked them, obviously those two are white. I'm 
thinking about the same concept as I have at the moment but more 
Adwaita-themed, so it wouldn't be a problem.

**Comment by stsdc** [comment]
If it is GTK, then it is native. Vala or C++ it doesn't matter.

**Comment by Tuxino88** [comment]
**I look forward to a first version with basic functions (voice and text messages).** 
deep hatred for Qt XD

**Comment by brneor** [comment]
> I have news!
> 
> First of all, sorry for being so absent from this project. Rather busy because University, some side-projects, personal life, and that feeling of being unable to make something good, but I hope that feeling will get the fuck off me soon 😄
> Second: It would be GTK-agnostic. As I said in another issue, I want it to be more "general" and less elementary-based. Most people use GTK but less people use Pantheon than people using GNOME. I do currently use GNOME as my daily-driven desktop.
> Third: I'm moving it from Vala to C++ (GTKMM), thus making it easier to use TdLib as it have native C++ libraries, and I feel quite comfortable using C++ rather than Vala. I'll upload ASAP the basis of the new Aircraft.
> Also: if I feel this is going somewhere good I'll think about funding. But only if this get at least a MVP, not before. I don't like the idea of being a ripoff.
> 
> PS: I'm so bad at designing, if this works I'll try to make an Adwaita-like icon, but I suck with vectorization, coloring and the likes.

I appreciate your work and I really liked the idea. Official Telegram client has been a pita for me in the past months. I'm not fluent in C++ but I have some free time and I'd love to help in some way.

**Comment by Tuxino88** [comment]
> > I have news!
> > First of all, sorry for being so absent from this project. Rather busy because University, some side-projects, personal life, and that feeling of being unable to make something good, but I hope that feeling will get the fuck off me soon smile
> > Second: It would be GTK-agnostic. As I said in another issue, I want it to be more "general" and less elementary-based. Most people use GTK but less people use Pantheon than people using GNOME. I do currently use GNOME as my daily-driven desktop.
> > Third: I'm moving it from Vala to C++ (GTKMM), thus making it easier to use TdLib as it have native C++ libraries, and I feel quite comfortable using C++ rather than Vala. I'll upload ASAP the basis of the new Aircraft.
> > Also: if I feel this is going somewhere good I'll think about funding. But only if this get at least a MVP, not before. I don't like the idea of being a ripoff.
> > PS: I'm so bad at designing, if this works I'll try to make an Adwaita-like icon, but I suck with vectorization, coloring and the likes.
> 
> I appreciate your work and I really liked the idea. Official Telegram client has been a pita for me in the past months. I'm not fluent in C++ but I have some free time and I'd love to help in some way.

I think the project has been abandoned, if you want to fork it and make it minimally usable we will be happy :D

**Comment by ccnmxns** [comment]
Sorry for being absent in this project :(

My daily driver is Mac, not Linux, and it’s quite abandoned. Of course it could be forked, I just did some basis as I think it would be suitable to use Telegram’s TD with C++ (GTKMM). But I’m quite busy and not using Linux and it’s sad but I couldn’t get this project to lift off finally, so feel free to fork it and make it great.

Just keep in mind my will of this application being a GTK app and not Elementary-based. Of course it could be usable in Elementary and playing nice with Elementary style but most users do use GNOME instead, so Adwaita should be the default style.

Of course you could do however you like, it would be your fork. But please put your efforts in ONE fork, not many, as it would be awesome to see people putting efforts in one project, who knows, maybe Telegram would like this.

This project made me feel that my contributions are being appreciated, but the problem is that I don’t feel on keeping it up as I wouldn’t be using it on this very day, as I don’t use GNOME.

My sincerest apologies on being away and hope someone would fork this project and make something superb.

Carlos

**Comment by stsdc** [comment]
Well, life happens! Such project needs 2-3 people to maintain and develop and a lot of time to spend to make it at least to a basic MVP.
I do have [a similar project ](https://github.com/stsdc/telefax/tree/dev), (@Suzamax you might recall 😅️), but written in Vala. (to see what Vala is capable of look at [Akira project](https://github.com/akiraux/Akira)).

And I agree with @Suzamax, such project should not have any elementary specific dependencies to cover as much as possible of GTK-based DE. Although it can use [elementary HIG](https://elementary.io/docs/human-interface-guidelines#human-interface-guidelines), it will also look nice in Adwaita or Pop_OS.

In conclusion, C++ or Vala - it doesn't matter as long as such an application will integrate into Gnome and Pantheon environment, will use native freedesktop features etc.

**Comment by Tuxino88** [comment]
> > I have news!
> > First of all, sorry for being so absent from this project. Rather busy because University, some side-projects, personal life, and that feeling of being unable to make something good, but I hope that feeling will get the fuck off me soon smile
> > Second: It would be GTK-agnostic. As I said in another issue, I want it to be more "general" and less elementary-based. Most people use GTK but less people use Pantheon than people using GNOME. I do currently use GNOME as my daily-driven desktop.
> > Third: I'm moving it from Vala to C++ (GTKMM), thus making it easier to use TdLib as it have native C++ libraries, and I feel quite comfortable using C++ rather than Vala. I'll upload ASAP the basis of the new Aircraft.
> > Also: if I feel this is going somewhere good I'll think about funding. But only if this get at least a MVP, not before. I don't like the idea of being a ripoff.
> > PS: I'm so bad at designing, if this works I'll try to make an Adwaita-like icon, but I suck with vectorization, coloring and the likes.
> 
> I appreciate your work and I really liked the idea. Official Telegram client has been a pita for me in the past months. I'm not fluent in C++ but I have some free time and I'd love to help in some way.

Look this 

https://github.com/melix99/telegrand
